pkg/sources/git/git_test.go

```diff
@@ -315,3 +406,51 @@ func TestSource_Chunks_Edge_Cases(t *testing.T) {
 		})
 	}
 }
+
+func TestPrepareRepo(t *testing.T) {
+	tests := []struct {
+		uri    string
+		path   bool
+		remote bool
+		err    error
+	}{
+		{
+			uri:    "https://github.com/dustin-decker/secretsandstuff.git",
+			path:   true,
+			remote: true,
+			err:    nil,
+		},
+		{
+			uri:    "file:///path/to/file.json",
+			path:   true,
+			remote: false,
+			err:    nil,
+		},
+		{
+			uri:    "no bueno",
+			path:   false,
+			remote: false,
+			err:    fmt.Errorf("unsupported Git URI: no bueno"),
+		},
+	}
+
+	for _, tt := range tests {
+		repo, b, err := PrepareRepo(tt.uri)
+		var repoLen bool
+		if len(repo) > 0 {
+			repoLen = true
+		} else {
+			repoLen = false
+		}
+		if repoLen != tt.path || b != tt.remote {
+			t.Errorf("PrepareRepo(%v) got: %v, %v, %v want: %v, %v, %v", tt.uri, repo, b, err, tt.path, tt.remote, tt.err)
+		}
+	}
+}
+
+func BenchmarkPrepareRepo(b *testing.B) {
+	uri := "https://github.com/dustin-decker/secretsandstuff.git"
+	for i := 0; i < b.N; i++ {
+		_, _, _ = PrepareRepo(uri)
```

Comment:
I think this has the possibility of being optimized out. We may want to check at least one return value to be safe.